base.docker

```diff
@@ -2,15 +2,20 @@ FROM python:3.6.1-slim
 
 ENV APP_DIR /app
 
-ENV NODE_VERSION 6.10.3
+ENV NODE_VERSION 6.12.2
+ENV YARN_VERSION 1.3.2-1
 
 RUN apt-get update && \
     apt-get install -y --no-install-recommends nginx \
                     libpcre3-dev libpq-dev libffi-dev libxml2-dev libxslt-dev \
-                    gcc make git curl xz-utils python2.7 python-pip && \
+                    gcc make git curl xz-utils python2.7 python-pip apt-transport-https && \
     curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.xz" && \
     tar -xJf "node-v$NODE_VERSION-linux-x64.tar.xz" -C /usr/local --strip-components=1 && \
     rm "node-v$NODE_VERSION-linux-x64.tar.xz" && \
+    curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+    echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
```

Comment:
My debian knowledge is poor so I may be barking up a wrong tree here.

I've been reading about `deb`: https://wiki.debian.org/SourcesList

The `stable` distribution of debian is `stretch` or `DebianStretch` i.e. v9.3 (https://wiki.debian.org/DebianStable). I think we are running Jessie (https://wiki.debian.org/DebianJessie) v8.8 according to python:3.6.1-slim docker file (https://github.com/docker-library/python/blob/a1aa406bfd8c7b129e6e0ee0ba972b863624ac0d/3.6/jessie/slim/Dockerfile) and @Wynndow checked by SSHing in to a box.

Therefore should we be doing `deb https://dl.yarnpkg.com/debian/ jessie main`?

All i've spotted is that thing might not be matching. I don't actually know/understand what the implications would be if they don't (I'll try and do a bit more reading this afternoon).

Reply:
https://github.com/yarnpkg/releases

not sure if they even offer a non `stable` distribution

Reply:
I guess if it all builds and works then it's OK?

Reply:
Yeah it doesn't seem like they offer an alternative but it builds and works :man_shrugging:

Reply:
I've tried changing `stable` to `jessie`. It didn't like it. 404.

Reply:
This seems to suggest that you shouldn't install packages for releases other than the one you're running...
https://wiki.debian.org/DontBreakDebian#Don.27t_make_a_FrankenDebian

Reply:
So should I just go back to using the install script?

Reply:
Other option is to update the base image? python have a `python:3.6.4-slim-stretch` tagged release, which may also make Louise happy?

Reply:
I was planning to update the base image as well, seems like a good time to do this

Reply:
OK! It seems to build and work now. There's an error when installing supervisor because we don't have one of its dependencies but according to Rob we don't actually need it, so not a concern.

Reply:
Have you tried pulling this in and building the other apps using it?

Reply:
Yeah, it's broken. 😢

Reply:
It _should_ work now

Reply:
Works for me.